core/src/main/java/org/apache/calcite/sql2rel/ConvertToChecked.java

```diff
@@ -50,13 +51,18 @@ public ConvertToChecked(RexBuilder builder) {
    * Visitor which rewrites an expression tree such that all
    * arithmetic operations that produce numeric values use checked arithmetic.
    */
-  static class ConvertRexToChecked extends RexShuttle {
+  class ConvertRexToChecked extends RexShuttle {
     private final RexBuilder builder;
 
     ConvertRexToChecked(RexBuilder builder) {
       this.builder = builder;
     }
 
+    @Override public RexNode visitSubQuery(RexSubQuery subQuery) {
+      RelNode result = subQuery.rel.accept(ConvertToChecked.this);
+      return subQuery.clone(result);
```

Comment:
why `clone`?

Reply:
This is not the Java Object.clone() method, that one has no arguments.

You can see the JavaDoc for the clone methods for RexCall in RexCall.clone().
```
  /**
   * Creates a new call to the same operator with different operands.
   *
   * @param type     Return type
   * @param operands Operands to call
   * @return New call
   */
  public RexCall clone(RelDataType type, List<RexNode> operands) {
```

Reply:
thanks

Reply:
If `result==subQuery.rel`, do we still need to clone it?

Reply:
Sorry, didn't see this until too late, if this is redundant I will fix it in a subsequent PR.
Have to find out whether clone makes the check or not.